improves compatibility with LLVM readelf, which doesn't implement the -w option
instead use to sed to find whether it's elf 32 or 64, then set pointer size accordingly

diff --git a/Makefile b/Makefile
index f0841aa..a5f9afc 100644
--- a/Makefile
+++ b/Makefile
@@ -43,9 +43,7 @@ NATIVE_ENDIAN=$(shell ${READELF} -h native.o \
 NATIVE_ENDIAN_LITTLE=$(findstring $(NATIVE_ENDIAN),$(findstring little,$(NATIVE_ENDIAN)))
 NATIVE_ENDIAN_BIG=$(findstring $(NATIVE_ENDIAN),$(findstring big,$(NATIVE_ENDIAN)))
 NATIVE_ENDIAN_VAL=$(if $(NATIVE_ENDIAN_LITTLE),ELFDATA2LSB,$(if $(NATIVE_ENDIAN_BIG),ELFDATA2MSB,ELFDATANONE))
-NATIVE_POINTER_SIZE=$(shell readelf -wi native.o \
-	| grep "Pointer Size:" \
-	| sed 's/.*: *//')
+NATIVE_POINTER_SIZE=$(shell ${READELF} --file-header native.o | sed -nE 's/^\s+Class:\s+([a-zA-Z1-9]+)/\1/ ; s/ELF32/4/p ; s/ELF64/8/p')
 %.o: %.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $< $(CFLAGS_FOR_SOURCE)
 install: dwz
